### PR TITLE
Added Vaccine and Cost Preference

### PR DESCRIPTION
**Pull request by prayuj**
This relates to Issue[#10](https://github.com/sushrut111/cowin-automation-extn/issues/10)

### CODE REVIEW

**Comment by prayuj** [comment]
@sushrut111 , Could you review the code?

File: contentScript.js [hunk 1]
```diff
@@ -419,9 +441,27 @@ const createForm = () => {
...
+  let costpreferenceid = "costpreference";
+  let costPreferenceSelector = createSelectInput(costpreferenceid, "", costPreferenceValue)
+  let free = createSelectOptions('select-free', 'Free', 'Free', costPreferenceValue === 'Free')
+  let paid = createSelectOptions('select-paid', 'Paid', 'Paid', costPreferenceValue === 'Paid')
+  costPreferenceSelector.appendChild(free);
+  costPreferenceSelector.appendChild(paid);
+  let costpreferencelabel = createLabel('costpreferencelabel', costpreferenceid, "Select Cost preference", textLabelStyles);
+
+  let vaccinepreferenceid = "vaccinepreference";
+  let vaccinePreferenceSelector = createSelectInput(vaccinepreferenceid, "", vaccinePreferenceValue)
```
Comment (sushrut111) [inline contentScript.js:456]: This way works too. But I was expecting more of checkbox like thing. 
Have you seen the cowin portal, 
Screenshot 2021-05-22 at 11 29 39 AM
Here I can select multiple filters

Comment (prayuj) [reply contentScript.js:456]: Ahh okay, I was going to create an issue later to make something like this
I will do it in this PR 👍

**Comment by sushrut111** [comment]
Selected buttons are not getting unselected on clicking again

**Comment by sushrut111** [comment]
> Selected buttons are not getting unselected on clicking again.
> 
> * On bigger viewport, it is an illusion because, the on hover color is same as button selected color - don't let color to be changed on hover
> * On small viewport, where touch simulation is enabled, you have to click inside and then again outside to unselect the button

This is fixed now, thanks! The small viewport is fixed to - it was because when we click outside the focus is taken off from the button - the same on hover stuff!

File: contentScript.js [hunk 2]
```diff
@@ -12,9 +50,12 @@ let ageSelectorText = window.localStorage.getItem("age");
 let searchByDistrictFlag = window.localStorage.getItem("searchpref") === "district" ? true : false;
 let keeptryingcontinuously = window.localStorage.getItem("keeptryingcontinuously") === "true" ? true : false;
 let timeslotind = window.localStorage.getItem("timeslot");
+let costPreferenceValue = window.localStorage.getItem("costpreference");
+let vaccinePreferenceValue = window.localStorage.getItem("vaccinepreference");
```
Comment (sushrut111) [inline contentScript.js:54]: This is of no use right?

File: contentScript.js [hunk 3]
```diff
@@ -509,12 +572,21 @@ const bindSubmitButtonToSaveInfo = () => {
...
     enableAutoRefresh = document.getElementById("enableautorefresh").checked;
     enableautoconfirm = document.getElementById("enableautoconfirm").checked;
-    ageSelectorText = document.getElementById("ageselect").value;
     let searchPreftext = document.getElementById("searchpref").value;
     first_5_pin_digits = document.getElementById("pincodeinput").value;
     timeslotind = document.getElementById("timeslotinput").value;
     center_prefs_string = document.getElementById("centerprefinput").value;
     minavailability = document.getElementById("minavailabilityinput").value;
+    let temp_checked_buttons = [];
+    selected_button_checkbox = []
```
Comment (sushrut111) [inline contentScript.js:581]: this is a list (array)

File: contentScript.js [hunk 4]
```diff
@@ -526,9 +598,12 @@ const bindSubmitButtonToSaveInfo = () => {
     window.localStorage.setItem("searchpref", searchPreftext);
     window.localStorage.setItem("pincode", first_5_pin_digits);
     window.localStorage.setItem("timeslot", timeslotind);
+    window.localStorage.setItem("costpreference", costPreferenceValue);
+    window.localStorage.setItem("vaccinepreference", vaccinePreferenceValue)
     window.localStorage.setItem("centerprefs", center_prefs_string);
     window.localStorage.setItem("minavailability", minavailability);
     window.localStorage.setItem("autoconfirm", enableautoconfirm);
+    window.localStorage.setItem("selectedbuttoncheckboxes", selected_button_checkbox)
```
Comment (sushrut111) [inline contentScript.js:606]: does setItem accept an array?

File: contentScript.js [hunk 5]
```diff
@@ -509,12 +572,21 @@ const bindSubmitButtonToSaveInfo = () => {
...
     center_prefs_string = document.getElementById("centerprefinput").value;
     minavailability = document.getElementById("minavailabilityinput").value;
+    let temp_checked_buttons = [];
+    selected_button_checkbox = []
+    for (const key in buttonCheckboxMapping) {
+      let button_checkbox = document.getElementById(key);
+      buttonCheckboxMapping[key].checked = button_checkbox.checked;
+      if (button_checkbox.checked) {
+        selected_button_checkbox.push(key)
+        temp_checked_buttons.push(buttonCheckboxMapping[key].label)
```
Comment (sushrut111) [inline contentScript.js:587]: what are we doing with `temp_checked_buttons`

**Review by sushrut111 — CHANGES_REQUESTED** [review]

File: contentScript.js (hunk 4 above)
Comment (prayuj) [reply contentScript.js:606]: setItem converts array into comma separated strings. I will use JSON.stringify to store the array, that seems better

File: contentScript.js (hunk 5 above)
Comment (prayuj) [reply contentScript.js:587]: Yikes! Will remove this!

File: contentScript.js (hunk 4 above)
Comment (sushrut111) [reply contentScript.js:606]: Yeah, that would be great, un-typed bugs should be avoided first hand!

**Review by sushrut111 — APPROVED** [review]
I will merge this after going through the flow once

**Comment by sushrut111** [comment]
@prayuj 

Screenshot 2021-05-23 at 12 09 01 AM

File: custom.css [hunk 6]
```diff
@@ -0,0 +1,5 @@
+.btn-outline-primary:hover {
```
Comment (sushrut111) [inline custom.css:1]:
```suggestion
.btn-outline-primary {
    background-color: white;
    color: rgb(13, 110, 253);
    border-color: rgb(13, 110, 253);
}
.btn-outline-primary:hover {
```

**Comment by sushrut111** [comment]
It gets back to its original behaviour on inner pages.

**Comment by prayuj** [comment]
> 
> 
> @prayuj
> Hover is still showing blue color
> 
> Screenshot 2021-05-23 at 12 09 01 AM

I think this is happening because the website's stylesheets are interfering with the css of the extension.
The modal too, if you notice becomes smaller on the inner pages! (Refer attached screenshot, Red box is Cowin's css and Green box is the extension's)
This is happening with the buttons too!
We have to figure out a way to use custom class names for our bootstrap css.

Capture

**Comment by sushrut111** [comment]
Hmm that's right.
For time being, I would suggest, we should have check boxes instead of buttons, if you are okay with that.

**Comment by prayuj** [comment]
> 
> 
> Hmm that's right.
> For time being, I would suggest, we should have check boxes instead of buttons, if you are okay with that.

Give me a couple of days, I'll try figuring this out!

**Comment by sushrut111** [comment]
> > Hmm that's right.
> > For time being, I would suggest, we should have check boxes instead of buttons, if you are okay with that.
> 
> Give me a couple of days, I'll try figuring this out! If not, I'll revert them to checkboxes

Sure. Can you just merge main branch into this?
If you do that, I will merge this PR and create a pre-release. Many people have been asking for the filters.

**Review by sushrut111 — APPROVED** [review]

**Pull request merged**